### PR TITLE
[TESTS] Do not save device stats in InfluxDB if database not provided

### PR DESCRIPTION
**Pull request by lukaszfryc**
Follow up on https://github.com/status-im/status-react/pull/6692. Do not send results to InfluxDB if db credentials not provided.

status: ready

### CODE REVIEW

**Review by yevh-berdnyk — APPROVED** [review]

File: test/appium/tests/conftest.py [hunk 1]
```diff
@@ -190,6 +190,14 @@ def pytest_unconfigure(config):
             pull.create_issue_comment(github_report.build_html_report(testrail_report.run_id))
 
 
+def should_save_device_stats(config):
+    return config.getoption('stats_db_host') and \
+           config.getoption('stats_db_port') and \
+           config.getoption('stats_db_username') and \
+           config.getoption('stats_db_username') and \
```
Comment (hetvart) [inline test/appium/tests/conftest.py]: twice `stats_db_username` option

File: test/appium/tests/conftest.py [hunk 2]
```diff
@@ -190,6 +190,14 @@ def pytest_unconfigure(config):
             pull.create_issue_comment(github_report.build_html_report(testrail_report.run_id))
 
 
+def should_save_device_stats(config):
+    return config.getoption('stats_db_host') and \
```
Comment (hetvart) [inline test/appium/tests/conftest.py]: do not overdrive the method with multi `and` verification.
you can use python `all()` built-in function for such checking:

```python
def should_save_device_stats(config):
    db_args = [config.getoption(option) for option in ('stats_db_host', 'stats_db_port', 'stats_db_username', 'stats_db_password')]
    return all(db_args)
```
`all()` returns True when all elements in the given iterable are true. If not, it returns False.

Comment (lukaszfryc) [reply test/appium/tests/conftest.py]: sure, we can sue `all()`. But, I find simple conjunction (`and`) much more readable in this case. There is no need to create an array (`db_args`). But, I can change it if you think we should use `all()`

Comment (hetvart) [reply test/appium/tests/conftest.py]: > sure, we can sue `all()`. But, I find simple conjunction (`and`) much more readable in this case. There is no need to create an array (`db_args`). But, I can change it if you think we should use `all()`

I proposed just more pythonic way to do so - gold rule is when you have more than 2 `and/or` conditions. `all()` is written in C and does this verification faster then `and/or` syntax.

Regarding readability, well it depends mostly on developer preferences/style though.

**Review by hetvart — APPROVED** [review]

File: test/appium/tests/conftest.py (hunk 2 above)
Comment (lukaszfryc) [reply test/appium/tests/conftest.py]: thanks! I'll change it. I think I still need to learn how to code in a more pythonic way :)

**Pull request merged**